### PR TITLE
Add JMeter plan for Find

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context
Following the structure of the Apply and Manage plans, add a plan which
exercises common paths in Find.

## Changes proposed in this pull request
- Navigate through the three options for searching for a course,
  clicking on a course page each time.
- Add configurable wait factor and thread count.

## Guidance to review
- Diff should be adequate to get an overview of what paths are being hit
- Can be run locally without any additional dependencies (doesn't need any extra files or scripts like the Apply plan).

## Link to Trello card
https://trello.com/c/sj551ab3

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: jmeter/plans/find.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
+require 'ruby-jmeter'
+
+BASEURL = ENV.fetch('JMETER_TARGET_BASEURL', 'http://localhost:3002')
+WAIT_FACTOR = ENV.fetch('JMETER_WAIT_FACTOR', 1).to_f
+THREAD_COUNT = ENV.fetch('JMETER_THREAD_COUNT', 200)
```
Comment (malcolmbaig) [inline jmeter/plans/find.rb:5]: Looking at the Find data on the day of the last spike, the concurrent user count estimate was ~150 at the peak hour. I've set a default count of 200 here, reflecting the estimate plus some additional margin.

This is far lower than the ~900 users we're testing in the Apply plan, but can be adjusted via the env var to test at that level as well if needed.

Comment (davidgisbey) [reply jmeter/plans/find.rb:5]: Yea it seems very unlikely that apply will get 6 times the traffic Find got last year. Not that i think it hurts to be over prepared. Do you think it's worth ramping the load test up to say 300 if 200 causes no issues just to be safe?

**Comment by davidgisbey** [comment]
It's def worth @mnacos having a look as i'm not super familiar with Jmeter, but it looks good to me Malcolm

File: jmeter/plans/find.rb (hunk 1 above)
Comment (malcolmbaig) [reply jmeter/plans/find.rb:5]: @mnacos just occurred to me that we'll need changes to how this is executed to support the new thread count env var. Is there anything else that needs to happen in code other than the Dockerfile update I just pushed?

Comment (mnacos) [reply jmeter/plans/find.rb:5]: No, should be fine. You can pretty much define any env var you need via `cf set-env ...`.

**Review by mnacos — APPROVED** [review]
Looking good. Should be fine to merge. I guess we'll need to monitor the response codes when this runs against the load-test environment, to detect any potential data issues.

**Pull request merged**